File: core/src/main/java/com/hgianastasio/biblioulbrav2/core/settings/repository/SettingRepository.java
package com.hgianastasio.biblioulbrav2.core.settings.repository;

import com.hgianastasio.biblioulbrav2.core.base.boundaries.OnResultListener;
import com.hgianastasio.biblioulbrav2.core.settings.Setting;

import java.io.IOException;
import java.util.List;

/**
 * Created by heitor_12 on 03/06/17.
 */

public interface SettingRepository {
    void get(String key, OnResultListener<Setting> resultListener) throws IOException;
    void getAll(OnResultListener<List<Setting>> resultListener) throws IOException;
    boolean set(Setting setting);
    boolean isSeted(String key);
    boolean remove(String key);
}

File: core/src/main/java/com/hgianastasio/biblioulbrav2/core/settings/repository/SettingRepository.kt
package com.hgianastasio.biblioulbrav2.core.settings.repository

import com.hgianastasio.biblioulbrav2.core.base.boundaries.OnResultListener
import com.hgianastasio.biblioulbrav2.core.settings.Setting
import java.io.IOException

/**
 * Created by heitor_12 on 03/06/17.
 */
interface SettingRepository {
    @Throws(IOException::class)
    operator fun get(key: String?, resultListener: OnResultListener<Setting?>)

    @Throws(IOException::class)
    fun getAll(resultListener: OnResultListener<List<Setting>>)
    fun set(setting: Setting?): Boolean
    fun isSet(key: String?): Boolean
    fun remove(key: String?): Boolean
}